### PR TITLE
[GLUTEN-5309] Add UTs for Spark 3.5

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

Add UTs for Spark3.5. Note that this PR haven't fixed the failed UTs but only sure compilation successful.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by zhouyuan** [comment]
@yma11 I think you will need to add one entry for Spark 35 unit tests here, otherwise the code is not compiled or tested?
https://github.com/apache/incubator-gluten/blob/main/.github/workflows/velox_docker.yml

**Comment by yma11** [comment]
> @yma11 I think you will need to add one entry for Spark 35 unit tests here, otherwise the code is not compiled or tested? https://github.com/apache/incubator-gluten/blob/main/.github/workflows/velox_docker.yml

@zhouyuan I add a build step for it and will enable the tests after excluding the failed ones in next PR. Thanks.

File: .github/workflows/velox_docker.yml [hunk 1]
```diff
@@ -636,4 +636,53 @@ jobs:
...
+        run: |
+          cd $GITHUB_WORKSPACE/cpp && \
+          source /opt/rh/gcc-toolset-9/enable && \
+          ./compile.sh --build_velox_backend=ON --build_protobuf=ON --build_tests=ON --build_examples=ON --build_benchmarks=ON
+      - name: Setup java and maven
+        run: |
+          wget https://downloads.apache.org/maven/maven-3/3.8.8/binaries/apache-maven-3.8.8-bin.tar.gz
+          tar -xvf apache-maven-3.8.8-bin.tar.gz
+          mv apache-maven-3.8.8 /usr/lib/maven
+      - name: Prepare spark.test.home for Spark 3.5.0 (other tests)
```
Comment (zhouyuan) [inline .github/workflows/velox_docker.yml:663]: curious why use spark350 here? 
spark 351 has been released for a while

**Review by zhouyuan — APPROVED** [review]
just one question on spark 350.

File: .github/workflows/velox_docker.yml (hunk 1 above)
Comment (yma11) [reply .github/workflows/velox_docker.yml:663]: Sorry for not noticing this. Let me update it.

Comment (yma11) [reply .github/workflows/velox_docker.yml]: Done.

**Pull request merged**